src/Command/AbstractDumpCommand.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Phoenix\Command;
+
+use Comparator\StructureComparator;
+use Dumper\Dumper;
+use Dumper\Indenter;
+use Phoenix\Database\Element\MigrationTable;
+use Phoenix\Database\Element\Structure;
+use Phoenix\Exception\InvalidArgumentValueException;
+use Phoenix\Migration\MigrationCreator;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+abstract class AbstractDumpCommand extends AbstractCommand
+{
+    protected function configure(): void
+    {
+        $this
+            ->addOption('ignore-tables', null, InputOption::VALUE_REQUIRED, 'Comma separated list of tables to ignore (Structure and data).', 'phoenix_log')
+            ->addOption('indent', 'i', InputOption::VALUE_REQUIRED, 'Indentation. Available values: 2spaces, 3spaces, 4spaces, 5spaces, tab', '4spaces')
+            ->addOption('migration', null, InputOption::VALUE_REQUIRED, 'Name of migration', 'Initialization')
+            ->addOption('dir', null, InputOption::VALUE_REQUIRED, 'Directory to create migration in')
+            ->addOption('template', null, InputOption::VALUE_REQUIRED, 'Path to template')
+        ;
+
+        parent::configure();
+    }
+
+    protected function runCommand(): void
+    {
+        $indenter = new Indenter();
+        $indent = $indenter->indent($this->input->getOption('indent'));
+        $dumper = new Dumper($indent, 2);
+
+        $migration = $this->input->getOption('migration') ?: 'Initialization';
+        $migrationCreator = new MigrationCreator($migration, $indent, $this->input->getOption('template'));
+
+        $sourceStructure = $this->sourceStructure();
+        $targetStructure = $this->targetStructure();
+
+        $up = $this->createUpDown($sourceStructure, $targetStructure, $dumper, 'up', $this->shouldLoadData());
+        $down = $this->createUpDown($targetStructure, $sourceStructure, $dumper, 'down');
+
+        $dir = $this->input->getOption('dir');
+        $migrationDir = $this->chooseMigrationDir($dir);
+        $migrationPath = $migrationCreator->create($up, $down, $migrationDir);
+
+        $this->writeln('');
+        $this->writeln('<info>Migration "' . $migration . '" created in "' . $migrationPath . '"</info>');
+        $this->outputData['migration_name'] = $migration;
+        $this->outputData['migration_filepath'] = $migrationPath;
+    }
+
+    abstract protected function sourceStructure(): Structure;
+
+    abstract protected function targetStructure(): Structure;
+
+    abstract protected function shouldLoadData(): bool;
```

Comment:
Merge these methods to one - load data, in diff will return empty array and in dump will check if data are required and then load them or return empty array